### ISSUE TITLE
Dock transparency disappears after exiting the Overview

### ISSUE THREAD

**Issue by j-james** [home-sweet-gnome/dash-to-panel#1333]
**Describe the bug**
On GNOME 40, the panel's background turns to black regardless of opacity setting after exiting the Overview menu. Locking and unlocking the screen or disabling and enabling the extension temporarily fixes the problem.

This was the most common bug reported in #1303.

**Linux distribution and version**
Arch 5.12

**GNOME Shell version**
40.1

**Dash-to-Panel version**
42

**Where was Dash-to-Panel installed from?**
`gnome-shell-extension-dash-to-panel-git` from the AUR.

**Screenshots / Video captures**
Before entering the overview:
![2021-05-11-003412](https://user-images.githubusercontent.com/35242550/117777610-a0ec2f80-b1f1-11eb-8f95-f983baa18fa1.png)

After leaving the overview:
![2021-05-11-003417](https://user-images.githubusercontent.com/35242550/117777656-aba6c480-b1f1-11eb-9b99-ba436d254202.png)

**Additional Info**
A tremendous thank you to everyone involved in porting this extension to GNOME 40, especially @philippun1!

**Comment by Testing-XYZ** [comment]
Not sure how to fix this issue in extension but here is a workaround. Run the following command in bash at startup:

`gdbus call --session --dest org.gnome.Shell --object-path /org/gnome/Shell --method org.gnome.Shell.Eval 'Function.prototype.clone = function() {
    var that = this;
    var temp = function temporary() { return that.apply(this, arguments); };
    for(var key in this) {
        if (this.hasOwnProperty(key)) {
            temp[key] = this[key];
        }
    }
    return temp;
};
Main.overview._hideDoneClone = Main.overview._hideDone.clone();
Main.overview._hideDone = function() {
    Main.overview._hideDoneClone();
    Main.loadTheme();
};'`

Basically reloads theme after hiding overview and as a result there is a slight delay. The clone function comes from [here](https://stackoverflow.com/questions/1833588/javascript-clone-a-function). It's quite inelegant but I'm not familiar with Gjs.

**Comment by philippun1** [comment]
@Testing-XYZ @j-james  I made a pull request #1337 (yes, this is some leet shit ^^). Please check if this fixes the issue for you.

**Comment by j-james** [comment]
It does indeed! The panel does noticeably flash black when exiting the overview quickly, but I think that's more to do with a different bug (the slight jumpiness of the exit overview animation)

**Comment by j-james** [comment]
Hmm, strangely enough, this fix stops working after some time.

**Comment by philippun1** [comment]
@j-james I had this behaviour once, but I cannot reproduce it. Indeed strange. I will keep an eye on this.

**Comment by sc44** [comment]
Hello,
i used the Transparent Shell extension to fix this. 
But since yesterday i test the fix from @philippun1 and i have no problems. The fix runs for me. Thanks.

**Comment by j-zero** [comment]
I can confirm this bug on Arch Linux, Gnome Shell 40.1 and dash-to-panel-git from AUR

**Comment by KonstantMan** [comment]
I can confirm this bug on Arch Linux, Gnome Shell 40.1 and dash-to-panel-git from AUR

**Comment by jderose9** [comment]
It seems like this one was resolved by ec4e0784ae4a468ed7c5dc68693ae777f5cfe4c5. Let me know if it persists and I will re-open.

**Comment by j-james** [comment]
I am actually seeing a bunch of more bugs after updating on Wayland - the disappearing transparency remains, the overview has become unusable, and switching between the desktop and the overview is stuttery and flashes grey. I'll try and make sure this isn't an issue on my end, and get a recording this afternoon.

**Comment by jderose9** [comment]
Oh I see, yes this still seems to be broken in Wayland.

**Comment by j-james** [comment]
Here's a recording of the overview issue:

https://user-images.githubusercontent.com/35242550/121725654-6995ca80-ca9e-11eb-9e80-7f6d7a0305b6.mp4

**Comment by KonstantMan** [comment]
> It seems like this one was resolved by [ec4e078](https://github.com/home-sweet-gnome/dash-to-panel/commit/ec4e0784ae4a468ed7c5dc68693ae777f5cfe4c5). Let me know if it persists and I will re-open.

I can confirm this bug remains  on Arch Linux, Gnome Shell 40.1 X11 and dash-to-panel-git from AUR

**Comment by philippun1** [comment]
This bug seems to be fixed with 258083f. Tested on Ubuntu and Manjaro.

**Comment by KonstantMan** [comment]
> This bug seems to be fixed with [258083f](https://github.com/home-sweet-gnome/dash-to-panel/commit/258083f3e30ef6dfb23ad2b5580c05f0faba7e3f). Tested on Ubuntu and Manjaro.

Manjaro and Ubuntu don't support gnome-shell  40 yet 
so the bug remains 
Tested on Arch Linux, Gnome Shell 40.1 X11 and dash-to-panel-git from AUR

**Comment by jderose9** [comment]
I think this one is fixed, for real this time.

**Comment by philippun1** [comment]
@KonstantMan Please pull the latest master branch, the AUR package is outdated, and verify again.
Ubuntu with shemgp PPA has Gnome 40.1 and Manjaro testing has Gnome 40.2.

@jderose9 Sadly I just noticed that it does not work if you use the three finger swipe up gesture on wayland. Clicking on "Show Applications" and using the Super Key works fine though. Anyone can confirm this?

**Comment by j-james** [comment]
@jderose9 Unfortunately, I notice no difference on Wayland.

**Comment by j-james** [comment]
Something I didn't notice before is that this isn't just transparency styles, overriding the panel background color at all (even with a solid color) will trigger this bug.

**Comment by Testing-XYZ** [comment]
> Sadly I just noticed that it does not work if you use the three finger swipe up gesture on wayland. Clicking on "Show Applications" and using the Super Key works fine though. Anyone can confirm this?

I can confirm that gesture-triggered overview do not produce transparency. Tested on Arch and dash-to-panel-git from AUR.

**Comment by jderose9** [comment]
Ok, I will do the shameful double re-open :)

**Comment by jderose9** [comment]
Looks like in overview.js, it will need to be overridden somehow where the panel style is cleared. There's one instance in _gestureEnd and two in runStartupAnimation.
https://gitlab.gnome.org/GNOME/gnome-shell/-/blob/master/js/ui/overview.js#L396
https://gitlab.gnome.org/GNOME/gnome-shell/-/blob/master/js/ui/overview.js#L650
https://gitlab.gnome.org/GNOME/gnome-shell/-/blob/master/js/ui/overview.js#L671

The _gestureEnd is going to be a pain to override though with the way the signal is bound to that method on init.

**Comment by jderose9** [comment]
Ok, anyone want to give this new commit a whirl? Thanks!

**Comment by j-james** [comment]
Looks to be working!

**Comment by philippun1** [comment]
@jderose9 Can confirm it works now.

**Comment by philippun1** [comment]
Now there is
`JS ERROR: Extension dash-to-panel@jderose9.github.com: TypeError: can't redefine non-configurable property "style"`
after suspend and resume.
A simple try/catch fixes it, but I'm not sure if there are any drawbacks...

**Comment by j-james** [comment]
When I get that error, my bar looks like this: 

![2021-06-19-161057](https://user-images.githubusercontent.com/35242550/122657512-f84abd00-d118-11eb-8980-0d7bd6ce4a54.png)

**Comment by jderose9** [comment]
@philippun1 Made commit 6e50948a8f2c28baaf3bc1fdb74a42d870e0e971 before I saw your PR, sorry!

**Comment by j-james** [comment]
This issue hasn't shown up at all since `35081df`, so I'm closing this unless anyone else still has problems. Thanks to everyone who worked on fixing it!
